hack/test-codegen.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Copyright 2019 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+${scriptdir}/generate-internal-groups.sh \
+  all \
+  github.com/pingcap/tidb-operator/tests/pkg/apiserver \
+  github.com/pingcap/tidb-operator/tests/pkg/apiserver/apis \
+  github.com/pingcap/tidb-operator/tests/pkg/apiserver/apis \
+  "example.pingcap.com:v1alpha1,v1beta1" \
+  --go-header-file=${scriptdir}/boilerplate.go.txt
```

Comment:
we should verify the generated code  is up to date. I created a issue for this: https://github.com/pingcap/tidb-operator/issues/1110

Reply:
Thanks!